src/operator/numpy/np_tensordot_op-inl.h

```diff
@@ -0,0 +1,391 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_tensordot_op-inl.h
+ * \brief CPU Implementation of numpy-compatible tensordot
+ */
+
+
+#include "np_matrix_op-inl.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mxnet;
+using namespace mshadow;
+
+struct TensordotParam : public dmlc::Parameter<TensordotParam> {
+  mxnet::Tuple<int> a_axes_summed, b_axes_summed; 
+  DMLC_DECLARE_PARAMETER(TensordotParam) {
+    DMLC_DECLARE_FIELD(a_axes_summed);
+    DMLC_DECLARE_FIELD(b_axes_summed);
+  }
+};
+
+/**
+ * Gets matrix dimensions of a and b after transpose and reshape. 
+ */
+inline void GetMatrixDimensions(
+    int& ad1, 
+    int& ad2, 
+    int& bd1, 
+    int& bd2, 
+    const mxnet::Tuple<int>& a_axes_remained, 
+    const mxnet::Tuple<int>& a_axes_summed, 
+    const mxnet::Tuple<int>& b_axes_remained, 
+    const mxnet::Tuple<int>& b_axes_summed, 
+    const mxnet::TShape& a_shape, 
+    const mxnet::TShape& b_shape) {
+  ad1 = 1;
+  ad2 = 1;
+  bd1 = 1;
+  bd2 = 1;
+
+  for (int i = 0; i < a_axes_remained.ndim(); i++) {
+    ad1 *= a_shape[a_axes_remained[i]];
+  } 
+  for (int i = 0; i < a_axes_summed.ndim(); i++) {
+    ad2 *= a_shape[a_axes_summed[i]];
+  } 
+  for (int i = 0; i < b_axes_summed.ndim(); i++) {
+    bd1 *= b_shape[b_axes_summed[i]];
+  } 
+  for (int i = 0; i < b_axes_remained.ndim(); i++) {
+    bd2 *= b_shape[b_axes_remained[i]];
+  } 
+}
+
+/**
+ * gets new axes of a and b after transpose and reshape.
+ */
+inline void GetReorderedAxes(
+    const mxnet::Tuple<int>& a_axes_summed, 
+    mxnet::Tuple<int>& a_axes_remained, 
+    mxnet::Tuple<int>& a_axes, 
+    const mxnet::Tuple<int>& b_axes_summed,
+    mxnet::Tuple<int>& b_axes_remained, 
```

Comment:
Same here, pass by pointers if you want to modify it.

Reply:
Fixed. Thx.